pkg/controller/remotemachineset/remotemachineset_controller.go

```diff
@@ -321,12 +321,12 @@ func (c *Controller) syncClusterDeployment(key string) error {
 		return fmt.Errorf("error retrieving cluster object: %v", err)
 	}
 
-	clusterDeploymentStatus, err := controller.ClusterStatusFromClusterAPI(cluster)
+	clusterProviderStatus, err := controller.ClusterProviderStatusFromClusterAPI(cluster)
```

Comment:
I would rename the function `ClusterProviderStatusFromCluster`... there's no reason to distinguish cluster-api vs non-cluster-api clusters anymore